packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.types.ts

```diff
@@ -154,6 +154,22 @@ export interface IComboBoxProps extends ISelectableDroppableTextProps<IComboBox>
    */
   useComboBoxAsMenuWidth?: boolean;
 
+  /**
+   * Optional mode indicates if multi-choice selections is allowed.  Default to false
+   */
+  multiSelect?: boolean;
+
+  /**
+   * Keys that will be initially used to set selected items.
+   */
+  defaultSelectedKeys?: string[] | number[];
+
+  /**
+  * Keys of the selected items. If you provide this, you must maintain selection
+  * state by observing onChange events and passing a new value in when changed.
+  */
+  selectedKeys?: string[] | number[];

```

Comment:
After this ComboBox will have defaultSelectedKey, selectedKey, defaultSelectedKeys, and selectedKeys... having both *key and *keys seems weird... we should consider updating the *key props to also be string | number | string[] | number[] instead of duplicating the existing props

Reply:
Good point, but there are 2 potential concerns:
1. inconsistent with exiting IDropdownProps. If we change IDropdownProps, it will cause backward compat.
2. if we make *Key property store multiple values, it could be a bit surprising for some developers.

Reply:
Can't you just override the potential types inside of ComboBox.types.ts? If you do that, you won't be changing anything for dropdowns

Reply:
After trying, as much as I want to override the *Key props, it doesn't work.  According to the following, one can only override with a narrower type so that it ensures the sub-interface is still assignable to a reference of parent interface.  In our case, we are expanding instead of narrowing the type of *Key props.  So I'm going to leave this PR as is.  At least it is consistent with DropDown.

https://stackoverflow.com/questions/41285211/typescript-overriding-interface-property-type-defined-in-d-ts
https://github.com/Microsoft/TypeScript/issues/3402

Reply:
Okay. We should make sure to put mutuallyExclusive warnings for if you are using any of the *key props that you shouldn't also be using any of the *keys props

Reply:
Just a thought, couldn't you update the base and then restrict the types for dropdown? The only remaining downside at that point is that you don't know if you were given an array or an individual value (which I imagine we could always may the value an array in the code to get around this)

Reply:
OK. I'll make that change now if that's your preference. Just wanted to make sure you saw what I pointed out 4-replies above and feel comfortable about it.

Reply:
Just pushed changes. Please review and sign off. thx